pkg/ebpf/dns_cache.go

```diff
@@ -102,12 +112,18 @@ func (c *reverseDNSCache) Get(conns []ConnectionStats, now time.Time) map[util.A
 			return
 		}
 
-		if names := c.getNamesForIP(addr, expiration); names != nil {
-			resolved[addr] = names
+		if _, ok := oversized[addr]; ok {
 			return
 		}
 
-		unresolved[addr] = struct{}{}
+		names := c.getNamesForIP(addr, expiration)
+		if len(names) == 0 {
+			unresolved[addr] = struct{}{}
+		} else if len(names) >= c.maxDomainsPerIP {
+			oversized[addr] = struct{}{}
```

Comment:
I am wondering if this is the best approach. The `merge` function is already maintaining the constraint  `len(names) <= c.maxDomainsPerIP`. If that is the case, can we instead just return whatever domains we already have? Otherwise, we will be discarding _all_ the domains in the specific case `len(names) == c.maxDomainsPerIP` - and I am not sure if we should waste all that information. Am I missing something here?

Reply:
Talked with @leeavital offline. The rationale for preferring sending no data to sending incomplete data makes sense.  
Just one small thing - we can probably get rid of the `>`.